test/node_redis.spec.js

```diff
@@ -0,0 +1,675 @@
+                    client.once("connect", function () {
+                        client.flushdb(done)
+                    });
+                });
+
+                afterEach(function () {
+                    client.end();
+                });
+
+                describe("when redis closes unexpectedly", function () {
+                    it("reconnects and can retrieve the pre-existing data", function (done) {
+                        client.on("reconnecting", function on_recon(params) {
+                            client.on("connect", function on_connect() {
+                                async.parallel([function (cb) {
+                                    client.get("recon 1", function (err, res) {
+                                        helper.isString("one")(err, res);
+                                        cb();
+                                    });
+                                }, function (cb) {
+                                    client.get("recon 1", function (err, res) {
+                                        helper.isString("one")(err, res);
+                                        cb();
+                                    });
+                                }, function (cb) {
+                                    client.get("recon 2", function (err, res) {
+                                        helper.isString("two")(err, res);
+                                        cb();
+                                    });
+                                }, function (cb) {
+                                    client.get("recon 2", function (err, res) {
+                                        helper.isString("two")(err, res);
+                                        cb();
+                                    });
+                                }], function (err, results) {
+                                    client.removeListener("connect", on_connect);
+                                    client.removeListener("reconnecting", on_recon);
+                                    done(err);
+                                });
+                            });
+                        });
+
+                        client.set("recon 1", "one");
+                        client.set("recon 2", "two", function (err, res) {
+                            // Do not do this in normal programs. This is to simulate the server closing on us.
+                            // For orderly shutdown in normal programs, do client.quit()
+                            client.stream.destroy();
+                        });
+                    });
+
+                    describe("and it's subscribed to a channel", function () {
+                        // reconnect_select_db_after_pubsub
+                        // Does not pass.
+                        // "Connection in subscriber mode, only subscriber commands may be used"
+                        xit("reconnects, unsubscribes, and can retrieve the pre-existing data", function (done) {
```

Comment:
Going to try to get this one test working, just finishing porting the final methods first.